ewkcoffea/modules/objects_wwz.py

```diff
@@ -17,32 +17,40 @@ def get_cleaned_collection(obj_collection_a,obj_collection_b,drcut=0.4):
 ######### WWZ 4l analysis object selection #########
 
 # WWZ preselection for electrons
-def is_presel_wwz_ele(ele,tight):
+def is_presel_wwz_ele(ele,year,tight):
     mask = (
         (ele.pt               >  get_ec_param("wwz_pres_e_pt")) &
         (abs(ele.eta)         <  get_ec_param("wwz_pres_e_eta")) &
         (abs(ele.dxy)         <  get_ec_param("wwz_pres_e_dxy")) &
-        (abs(ele.dz)          <  get_ec_param("wwz_pres_e_dz")) &
+        (abs(ele.dz)          <  get_ec_param("wwz_pres_e_dz"))
+    )
+    mask_run2 = (
+        (ele.miniPFRelIso_all < get_ec_param("wwz_pres_e_miniPFRelIso_all")) &
         (abs(ele.sip3d)       <  get_ec_param("wwz_pres_e_sip3d")) &
-        (ele.miniPFRelIso_all <  get_ec_param("wwz_pres_e_miniPFRelIso_all")) &
         (ele.lostHits         <= get_ec_param("wwz_pres_e_lostHits"))
     )
-    if tight: mask = (mask & ele.convVeto & (ele.tightCharge == get_ec_param("wwz_pres_e_tightCharge")))
-    return mask
+    if ("2022" in year): mask_return = (mask & ele.mvaIso_WP80)
+    if ("2022" not in year): mask_return = (mask & mask_run2)
+    if tight: mask_return = (mask_return & ele.convVeto & (ele.tightCharge == get_ec_param("wwz_pres_e_tightCharge")))
```

Comment:
Is this "tight" mask is relevant and is wanted to be applied to R3?

Reply:
I can remove this as a v0 for Run3

Reply:
Thanks for updating that. However, the if logic here is now pretty hard to follow (for me anyway). I think it would first of all be good to just pass your `is2022` variable to this function so that we don't have to check if "2022" is in the year. 

Then secondly I think you can move the final if into the else (that went with the first if). So instead of:
```if ("2022" in year): 
        mask_return = (mask & ele.mvaIso_WP80 & ele.convVeto)
    else: 
        mask_return = (mask & mask_run2)
    if tight and ("2022" not in year): 
        mask_return = (mask_return & ele.convVeto & (ele.tightCharge == get_ec_param("wwz_pres_e_tightCharge")))
```
It would be something like this:
```
    if  is2022: 
        mask_return = (mask & ele.mvaIso_WP80 & ele.convVeto)
    else: 
        if tight: 
            mask_return = (mask_return & ele.convVeto & (ele.tightCharge == get_ec_param("wwz_pres_e_tightCharge")))
        else:
            mask_return = (mask & mask_run2)
```

Reply:
Though additionally, are you sure this does not change the behavior for R2? Where do you apply the mask_run2 in the case of tight?

Reply:
In fact, at this point, I think it might be cleaner to just split this function into two different functions. There is really not much overlapping code here (just the first 4 lines only).

Reply:
Actually, it seems we never use this function for R2 _without_ passing Tight=True. So let's just get rid of the tight option all together and always apply that for R2. That will make the function a lot cleaner, and I think we can leave it as one function. 

But please do pass the is2022 variable and restructure the if statements as described above.

Reply:
Thanks for the update. I think this looks a _lot_ easier to read now. One last request (sorry), could you remove the `year` argument now? Since it's no longer used by the function.